src/rgw/rgw_cr_rados.h

```diff
@@ -750,7 +750,10 @@ class RGWAsyncFetchRemoteObj : public RGWAsyncRadosRequest {
                                                       bucket_info(_bucket_info),
                                                       key(_key),
                                                       versioned_epoch(_versioned_epoch),
-                                                      copy_if_newer(_if_newer), zones_trace(_zones_trace) {}
+                                                      copy_if_newer(_if_newer)
+  {
+    zones_trace = *_zones_trace;
```

Comment:
i believe this argument was optional before, and this change makes it mandatory - which is fine, but i'd prefer to enforce this non-null requirement by taking the argument as `const rgw_zone_set&` instead of `rgw_zone_set*` in all 4 of these class constructors (RGWAsyncFetchRemoteObj/CR and RGWAsyncRemoveObj/CR). we'll have to change the signatures of RGWDataSyncModule and its overrides as well

does that make sense?

Reply:
Yeah `const rgw_zone_set&` sounds better,  but I think since only data sync need zone_trace, seems it's not worth of change the RGWDataSyncModule signatures. 
Maybe we can init empty or copy rgw_zone_set in `sync_object/remove_object`, then move in `RGWAsyncFetchRemoteObj/RGWAsyncRemoveObj`, will it look weird?

Reply:
maybe the simplest option is to keep support for nullptr here?
```c++
if (_zones_trace) {
  zones_trace = *_zones_trace;
}
```
i'm fine with your approach too - up to you!